lib/unauthorized.js

```diff
@@ -15,11 +15,11 @@ module.exports = function(app, config, logger) {
 
 function addWwwAuthenticateHeaders(req, res, domain, clientId, resourceServer, realm) {
   var v1WwwAuthenticateHeader = util.format('Bearer realm="%s", scope="client_id=%s service=%s"',
-    domain, clientId, req.protocol + "://" + req.hostname + req.baseUrl);
+    domain, clientId, "https://" + req.hostname + req.baseUrl);
 
   if (resourceServer) {
     var v2WwwAuthenticateHeader = util.format('Bearer realm="%s", authorization_uri="https://%s/oauth/token"', realm, domain);
-    return res.header('WWW-Authenticate', [v1WwwAuthenticateHeader, v2WwwAuthenticateHeader]);
+    return res.header('WWW-Authenticate', [v2WwwAuthenticateHeader, v1WwwAuthenticateHeader]);

```

Comment:
Order matters for some libraries. This is saying "We prefer v2 if you can provide that, otherwise we also support v1)